tests/revision_forward_test.go

```diff
@@ -195,12 +196,130 @@ func TestRevisionIsForwardedCorrectly_BrioEnablesOsakaInBlockProcessing(t *testi
+		Code:    code,
+		Nonce:   1,
+	}
+
+	// 1)  start net with Allegro
+
+	net := StartIntegrationTestNet(t, IntegrationTestNetOptions{
+		Upgrades: AsPointer(opera.GetAllegroUpgrades()),
+		Accounts: []makefakegenesis.Account{brioOnlyContract},
+	})
+
+	client, err := net.GetClient()
+	require.NoError(t, err)
+	defer client.Close()
+
+	blockBeforeUpgrade, err := client.BlockByNumber(t.Context(), nil)
+	require.NoError(t, err)
+
+	// 1.1)  contract cannot be executed before Brio
+
+	err = doRpcCall(client, brioOnlyContract.Address, []byte{0x00, 0xFF}, blockBeforeUpgrade.Hash())
+	require.ErrorContains(t, err, "execution unsuccessful", "expected eth_call to fail before Brio upgrade")
+	trace, err := doTraceCall(client, brioOnlyContract.Address, []byte{0x00, 0xFF}, blockBeforeUpgrade.Hash())
+	require.NoError(t, err, "expected trace_call to succeed even if execution fails")
+	require.Contains(t, trace["error"].(string), "invalid opcode", "expected invalid opcode error in trace_call before Brio upgrade")
+
+	// 2)  upgrade to Brio
+
+	type rulesType struct {
+		Upgrades struct{ Brio bool }
+	}
+	rulesDiff := rulesType{
+		Upgrades: struct{ Brio bool }{Brio: true},
+	}
+
+	UpdateNetworkRules(t, net, rulesDiff)
+	AdvanceEpochAndWaitForBlocks(t, net)
+
+	blockAfterUpgrade, err := client.BlockByNumber(t.Context(), nil)
+	require.NoError(t, err)
+
+	// 2.1)  contract can be executed after Brio
+
+	err = doRpcCall(client, brioOnlyContract.Address, []byte{0x00, 0xFF}, blockAfterUpgrade.Hash())
+	require.NoError(t, err, "expected eth_call to execute with Brio upgrade")
+	trace, err = doTraceCall(client, brioOnlyContract.Address, []byte{0x00, 0xFF}, blockAfterUpgrade.Hash())
+	require.NoError(t, err, "expected trace_call to execute with Brio upgrade")
+	_, failed := trace["error"]
+	require.False(t, failed, "did not expect error in trace_call after Brio upgrade")
+	require.Len(t, trace["calls"].([]any), 1, "expected one traced call entry from CLZ contract after Brio upgrade")
+
+	// 2.2)  expect rcp calls failing if using older than fork block
+
+	err = doRpcCall(client, brioOnlyContract.Address, []byte{0x00, 0xFF}, blockBeforeUpgrade.Hash())
+	require.ErrorContains(t, err, "execution unsuccessful", "expected eth_call to fail before Brio upgrade")
+	trace, err = doTraceCall(client, brioOnlyContract.Address, []byte{0x00, 0xFF}, blockBeforeUpgrade.Hash())
+	require.NoError(t, err, "expected trace_call to succeed even if execution fails")
+	require.Contains(t, trace["error"].(string), "invalid opcode", "expected invalid opcode error in trace_call before Brio upgrade")
+}
+
+func doTraceCall(client *PooledEhtClient, contractAddress common.Address, callData []byte, blockHash common.Hash) (map[string]any, error) {
+	// debug_traceCall serves to test functions using StateTransition RPC method
+	config := map[string]any{
+		"tracer": "callTracer",
+	}
+	return InvokeRpcCallMethod[map[string]any]("debug_traceCall", contractAddress, client, blockHash, config)
```

Comment:
if you cast the result like so:
```Go
	result, err := InvokeRpcCallMethod("debug_traceCall", contractAddress, client, blockHash, config)
	return result.(map[string]any), err 
```
then you can drop the generalization from `InvokeRpcCallMethod` and just use `(any, error)` for the return type.